### PR TITLE
[UI] Homework page #140

### PR DESCRIPTION
**Pull request by timt1028**
#140 

截圖 2021-07-16 上午12 35 47
截圖 2021-07-16 上午12 35 57
截圖 2021-07-16 上午12 36 05

### CODE REVIEW

File: src/app/homework/HomeworkPageNavigationBar.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import {InPageNavigationBar} from "../commons/IndexBanner/InPageNavigationBar";
+import {AiFillYoutube} from "react-icons/all";
+
+const HomeworkPageNavigationBar = function ({currentURL, homeworkName, homeworkId}) {
```
Comment (Wally5077) [inline src/app/homework/HomeworkPageNavigationBar.js:4]: change to the `arrow function expression`

File: src/app/homework/progress/HomeworkContext.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import {createContext, useContext} from "react";
+
+export const HomeworkContext = createContext(null);
+
+export const useHomeworkContext = () => {
+    return useContext(HomeworkContext)
```
Comment (Wally5077) [inline src/app/homework/progress/HomeworkContext.js:6]: ;

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

File: src/app/homework/progress/HomeworkProgressPage.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                        width="1000px"
+                        tableHeaders={[
+                            <TableCell>Name</TableCell>,
+                            ...currentHomework.problemIds.map(problemId => <TableCell>{problemId}</TableCell>),
+                            <TableCell>Table Score</TableCell>
+                        ]}
+
+                        showFilterSearchBar={false}
+                    />
+                    <TextareaModal title={"Import Students"}
```
Comment (nyngwang) [inline src/app/homework/progress/HomeworkProgressPage.js:64]: Remove the `{}` around `""`

File: src/app/homework/progress/HomeworkProgressPage.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                                       description: "To see the homework progress for each students.",
+                                       Icon: AiOutlineMail,
+                                       placeholder: "studentA@example.com\nstudentB@example.com",
+                                       remark: "＊One student email per line.",
+                                       buttonName: "Import"
+                                   }}
+                                   show={showAddStudentModal}
+                                   onClose={() => setShowAddStudentModal(false)}
+                    />
+                    <TextareaModal title={"Import Groups"}
```
Comment (nyngwang) [inline src/app/homework/progress/HomeworkProgressPage.js:75]: ditto.

**Review by nyngwang — CHANGES_REQUESTED** [review]
Review done.

File: src/app/homework/HomeworkList.js [hunk 5]
```diff
@@ -4,20 +4,28 @@ import {ThreeDotsButton} from "../commons/buttons/ThreeDotsButton";
...
     const [showDeleteHomeworkConfirmationModal, setShowDeleteHomeworkConfirmationModal] = useState(false);
-    const refetchHomework = useCallback(() => {
+    const refetchHomework = useCallback((homeworkId) => {
         homeworkService.getAllHomework()
-            .then(setHomeworkList)
-    }, [setHomeworkList]);
+            .then(homeworkList => {
+                setHomeworkList(homeworkList);
+                setCurrentHomework(homeworkList.find(homework => parseInt(homework.id) === parseInt(homeworkId)));
+            })
```
Comment (Johnny850807) [inline src/app/homework/HomeworkList.js:27]: semicolon

**Review by Johnny850807 — APPROVED** [review]
LGTM

File: src/app/homework/HomeworkList.js [hunk 6]
```diff
@@ -51,47 +59,61 @@ const HomeworkList = () => {
...
-                                  key: homework => homework.id,
-                                  data: homework => [
-                                      <TableCell>{homework.id}</TableCell>,
-                                      <FakeLink>{homework.name}</FakeLink>,
-                                      <TableCell>{homework.problemIds.join(', ')}</TableCell>,
-                                      <TableCell>{actionItemsButton({homework})}</TableCell>,
-                                  ]
-                              }}/>
+        <>{shouldRedirect ?
+            <Redirect to={`/homework/${currentHomework.id}/homework`}/> : ""}
```
Comment (Wally5077) [inline src/app/homework/HomeworkList.js:63]: what's this? -> '/>'

File: src/app/homework/progress/HomeworkProgressPage.js [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+                    <div className="mt-4 mr-3" style={{width: "fit-content", textAlign: "center"}}>
+                        <div className="homework-button">
+                            <button className="import-students-button button mt-2"
+                                    onClick={() => setShowAddStudentModal(true)}>Import Students
+                            </button>
+                            <button className="import-groups-button button mt-2"
+                                    onClick={() => setShowAddGroupModal(true)}>Import Groups
+                            </button>
+                            <button className="button export-csv-button">Export CSV
+                            </button>
```
Comment (Wally5077) [inline src/app/homework/progress/HomeworkProgressPage.js:50]: button 有不止一個 property 的話，button 的 value 換行。(Ex1)
如果只有一個的話，就維持一行。(Ex2)
Ex1:
```Js
<button className="myClassName"
        onClick={() => doSomething()}>
        value
</button>
```
Ex2:
```Js
<button className="myClassName">value</button>
```

File: src/app/homework/progress/HomeworkProgressPage.js [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+                    <ItemListPage
+                        width="1000px"
+                        tableHeaders={[
+                            <TableCell>Name</TableCell>,
+                            ...currentHomework.problemIds.map(problemId => <TableCell>{problemId}</TableCell>),
+                            <TableCell>Table Score</TableCell>
+                        ]}
+                        showFilterSearchBar={false}
+                    />
```
Comment (Wally5077) [inline src/app/homework/progress/HomeworkProgressPage.js:62]: 放在上一行最後面

File: src/app/homework/progress/HomeworkProgressPage.js [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+                                   body={{
+                                       description: "To see the homework progress for each students.",
+                                       Icon: AiOutlineMail,
+                                       placeholder: "studentA@example.com\nstudentB@example.com",
+                                       remark: "＊One student email per line.",
+                                       buttonName: "Import"
+                                   }}
+                                   show={showAddStudentModal}
+                                   onClose={() => setShowAddStudentModal(false)}
+                    />
```
Comment (Wally5077) [inline src/app/homework/progress/HomeworkProgressPage.js:73]: ditto

File: src/app/homework/progress/HomeworkProgressPage.js [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+                                   body={{
+                                       description: "To see the homework progress for students in groups.",
+                                       Icon: AiOutlineUsergroupAdd,
+                                       placeholder: "group-name-1\ngroup-name-1",
+                                       remark: "＊One group name per line.",
+                                       buttonName: "Import"
+                                   }}
+                                   show={showAddGroupModal}
+                                   onClose={() => setShowAddGroupModal(false)}
+                    />
```
Comment (Wally5077) [inline src/app/homework/progress/HomeworkProgressPage.js:84]: ditto

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

**Review by nyngwang — APPROVED** [review]
LGTM.

File: src/app/homework/HomeworkList.js (hunk 6 above)
Comment (nyngwang) [reply src/app/homework/HomeworkList.js:63]: testing.

File: src/app/homework/progress/HomeworkProgressPage.js [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+                            <button className="import-students-button button mt-2"
+                                    onClick={() => setShowAddStudentModal(true)}>
+                                Import Students
+                            </button>
+                            <button className="import-groups-button button mt-2"
+                                    onClick={() => setShowAddGroupModal(true)}>
+                                Import Groups
+                            </button>
+                            <button className="button export-csv-button">Export CSV
+                            </button>
```
Comment (Wally5077) [inline src/app/homework/progress/HomeworkProgressPage.js:52]: one line

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done

**Review by Wally5077 — APPROVED** [review]
LGTM

**Pull request merged**